xnb/readers/Texture2DReader.py

```diff
@@ -1,108 +1,32 @@
 
 from __future__ import annotations
 from typing import List
-from PIL import Image
 
 from ..constants import SurfaceFormat
+from ..objects import Texture2D
 from ..streams import StreamIn
 from .Base import BaseReader
 
-import logging
-import numpy
-import io
-
 class Texture2DReader(BaseReader):
     def __init__(self, stream: StreamIn) -> None:
-        self.width: int | None = None
-        self.height: int | None = None
-        self.surface_format: SurfaceFormat | None = None
-        self.textures: List[bytes] = []
-        self.image_data: bytes = bytes()
-        self.logger = logging.getLogger(__name__)
+        self.texture: Texture2D | None = Texture2D()
```

Comment:
Either set the `self.texture` to `None`, or remove the `| None` from its type.

Reply:
true